src/Sylius/Bundle/AdminBundle/gulpfile.babel.js

```diff
@@ -0,0 +1,167 @@
+import yargs from 'yargs';
+
+const argv = yargs
+    .options({
+        rootPath: {
+            description: '<path> path to web assets directory',
+            type: 'string',
+            requiresArg: true,
+            required: true,
+        },
+        vendorPath: {
+            description: '<path> path to vendor directory',
+            type: 'string',
+            requiresArg: true,
+            required: false,
+        },
+        nodeModulesPath: {
+            description: '<path> path to node_modules directory',
+            type: 'string',
+            requiresArg: true,
+            required: true,
+        },
+    })
+    .argv
+;
+
+const env = process.env.GULP_ENV;
+const { rootPath, vendorPath = '', nodeModulesPath } = argv;
+const adminRootPath = rootPath + 'admin/';
+const vendorAdminPath = '' === vendorPath ? '' : vendorPath + 'AdminBundle/';
+const vendorUiPath = '' === vendorPath ? '../UiBundle/' : vendorPath + 'UiBundle/';
+
+const paths = {
+    admin: {
+        js: [
+            nodeModulesPath + 'jquery/dist/jquery.min.js',
+            nodeModulesPath + 'semantic-ui-css/semantic.min.js',
+            vendorUiPath + 'Resources/private/js/**',
+            vendorAdminPath + 'Resources/private/js/**',
+        ],
+        sass: [
+            vendorUiPath + 'Resources/private/sass/**',
+            vendorAdminPath + 'Resources/private/sass/**',
+        ],
+        css: [
+            nodeModulesPath + 'semantic-ui-css/semantic.min.css',
+            vendorUiPath + 'Resources/private/css/**',
+            vendorAdminPath + 'Resources/private/css/**',
+        ],
+        img: [
+            vendorUiPath + 'Resources/private/img/**',
+            vendorAdminPath + 'Resources/private/img/**',
+        ],
+    },
+};
+
+const sourcePathMap = [
+    {
+        sourceDir: path.relative('', vendorAdminPath + 'Resources/private/'),
+        destPath: '/SyliusAdminBundle/',
+    },
+    {
+        sourceDir: path.relative('', vendorUiPath + 'Resources/private/'),
+        destPath: '/SyliusUiBundle/',
+    },
+    {
+        sourceDir: path.relative('', nodeModulesPath),
+        destPath: '/node_modules/',
+    },
+];
+
+const mapSourcePath = function mapSourcePath(sourcePath, file) {
+    for (const { sourceDir, destPath } of sourcePathMap) {
+        if (sourcePath.substring(0, sourceDir.length) === sourceDir) {
+            return destPath + sourcePath.substring(sourceDir.length);
+        }
+    }
+
+    return sourcePath;
+};
+
+export const buildAdminJs = function buildAdminJs() {
+    return gulp.src(paths.admin.js, { base: './' })
+        .pipe(gulpif(env !== 'prod', sourcemaps.init()))
+        .pipe(concat('app.js'))
+        .pipe(gulpif(env === 'prod', uglify()))
+        .pipe(gulpif(env !== 'prod', sourcemaps.mapSources(mapSourcePath)))
+        .pipe(gulpif(env !== 'prod', sourcemaps.write('./')))
+        .pipe(gulp.dest(adminRootPath + 'js/'))
+        .pipe(livereload())
```

Comment:
I've added the `livereload()` call here because it seemed to be missing. Let me know if it was intentionally omitted.